diff --git CodeSniffer.php CodeSniffer.php
index 66c311d..f75bbd5 100644
--- CodeSniffer.php
+++ CodeSniffer.php
@@ -415,9 +415,9 @@ class PHP_CodeSniffer
             if ($installed !== null) {
                 $standard = $installed;
             } else if (is_dir($standard) === true
-                && is_file(realpath($standard.'/ruleset.xml')) === true
+                && is_file($standard.'/ruleset.xml') === true
             ) {
-                $standard = realpath($standard.'/ruleset.xml');
+                $standard = $standard.'/ruleset.xml';
             }
 
             if (PHP_CODESNIFFER_VERBOSITY === 1) {
@@ -545,7 +545,6 @@ class PHP_CodeSniffer
      */
     public function processRuleset($rulesetPath, $depth=0)
     {
-        $rulesetPath = realpath($rulesetPath);
         if (PHP_CODESNIFFER_VERBOSITY > 1) {
             echo str_repeat("\t", $depth);
             echo "Processing ruleset $rulesetPath".PHP_EOL;
@@ -654,7 +653,7 @@ class PHP_CodeSniffer
             if (in_array($sniff, $excludedSniffs) === true) {
                 continue;
             } else {
-                $files[] = realpath($sniff);
+                $files[] = $sniff;
             }
         }
 
@@ -796,10 +795,10 @@ class PHP_CodeSniffer
                 $newRef  = false;
                 $stdPath = $this->getInstalledStandardPath($stdName);
                 if ($stdPath !== null && $path !== '') {
-                    $newRef = realpath(dirname($stdPath).$path);
+                    $newRef = dirname($stdPath).$path;
                 }
 
-                if ($newRef === false) {
+                if (file_exists($newRef) === false) {
                     // The sniff is not locally installed, so check if it is being
                     // referenced as a remote sniff outside the install. We do this
                     // by looking through all directories where we have found ruleset
@@ -810,9 +809,9 @@ class PHP_CodeSniffer
                             continue;
                         }
 
-                        $newRef = realpath($dir.$path);
+                        $newRef = $dir.$path;
                         
-                        if ($newRef !== false) {
+                        if (file_exists($newRef) !== false) {
                             $ref = $newRef;
                         }
                     }
@@ -994,14 +993,20 @@ class PHP_CodeSniffer
                 continue;
             }
 
-            $slashPos = strrpos(substr($file, 0, $sniffPos), DIRECTORY_SEPARATOR);
+            if (strpos($file, 'phar://') === 0) {
+                $directorySeparator = '/';
+            } else {
+                $directorySeparator = DIRECTORY_SEPARATOR;
+            }
+
+            $slashPos = strrpos(substr($file, 0, $sniffPos), $directorySeparator);
             if ($slashPos === false) {
                 continue;
             }
 
             $className = substr($file, ($slashPos + 1));
             $className = substr($className, 0, -4);
-            $className = str_replace(DIRECTORY_SEPARATOR, '_', $className);
+            $className = str_replace($directorySeparator, '_', $className);
 
             // If they have specified a list of sniffs to restrict to, check
             // to see if this sniff is allowed.
@@ -2065,7 +2070,7 @@ class PHP_CodeSniffer
     {
         $installedPaths = self::getInstalledStandardPaths();
         foreach ($installedPaths as $installedPath) {
-            $path = realpath($installedPath.'/'.$standard.'/ruleset.xml');
+            $path = $installedPath.'/'.$standard.'/ruleset.xml';
             if (is_file($path) === true) {
                 return $path;
             }
